frankenscript/README.md

```diff
@@ -0,0 +1,191 @@
+
+The Docker image provides scripts to run individual files. The output will be
+written to the `./artifact` directory on your local machine.
+
+This will build the `./examples/01-objects.frank` script and generate a markdown file:
+
+```bash
+./fs.bash ./examples/01-objects.frank
+```
+
+You can use to generate the HTML version of the file. The HTML files load the
+markdown JS plugin from Internet and renders the figures as a single file
+interleaved with the lines of code that give rise to the subsequent figure.
+
+```bash
+/fs-html.bash ./examples/01-objects.frank
+```
+
+### Watching a file
+
+FrankenScript is intended to be used interactively. The Docker image provides a
+handy script to watch a file and rerun FrankenScript when the file is updated.
+
+We recommend that you create a `watch.frank` file in the `./artifact` directory
+of your local machine. In the Docker container you can then run:
+
+```bash
+./watch.bash ./output/watch.frank 
+```
+
+This will generate a `watch.md` file. VS Code will automatically reload the file
+when it has been updated by FrankenScript.
+
+Happy Hacking!
+
+### Run all examples
+
+The Docker image provides a script to run all examples from the `examples`
+directory of FrankenScript:
+
+```bash
+./all-examples.bash
+```
+
+This will generate several Markdown files in the output directory of the Docker
+container. This is mapped to the `./artifact` directory on your local machine.
+
+If you want to generate the HTML files for all examples, you can use the following
+alternative script:
+
+```bash
+./all-examples-html.bash
+```
+
+### Recreate the figures from the paper
+
+The artifact contains scripts to recreate the figures from the paper. These are found
+in the `./figures` directory. All of them can be recreated with the following command:
+
+```bash
+./all-figures.bash
+```
+
+This will generate several Markdown files in the output directory of the Docker
+container. This is mapped to the `./artifact` directory on your local machine.
+
+If you want to generate the HTML files for all figures, you can use the following
+alternative script:
+
+```bash
+./all-figures-bash.bash
+```
+
+## Troubleshooting
+
+This section covers some issues you might encounter while building or working with the artifact:
+
+* **"Permission denied" when running Docker commands:**
+
+    Dockers documentation explains how you can manage Docker as a non-root user:
+    <https://docs.docker.com/engine/install/linux-postinstall/>
+    You can also invoke all Docker commands as root, that should work just fine
+    for FrankenScript.
+
+* **Error when running FrankenScript in Docker:
+  "Unable to open output file: /artifact/output/res/XYZ.md"**
+
+    The `frankenscript` user might have lost permissions to write into the
+    `./output` directory when it was mounted. You can manually allow access
+    again with this command in Docker:
+
+    ```bash
+    sudo chown frankenscript -R /artifact/output
+    ```
```

Comment:
Is this still correct?  I think we are using the ubuntu user now?

Reply:
The user was outdated, very good catch!